packages/broker/src/storage/StoragePlugin.ts

```diff
@@ -0,0 +1,91 @@
+import { router as dataQueryEndpoints } from './DataQueryEndpoints'
+import { router as dataMetadataEndpoint } from './DataMetadataEndpoints'
+import { router as storageConfigEndpoints } from './StorageConfigEndpoints'
+import { Plugin, PluginOptions } from '../Plugin'
+import { StreamFetcher } from '../StreamFetcher'
+import { Storage, startCassandraStorage } from './Storage'
+import { StorageConfig } from './StorageConfig'
+import { StreamPart } from '../types'
+import { Wallet } from 'ethers'
+import PLUGIN_CONFIG_SCHEMA from './config.schema.json'
+
+export interface StoragePluginConfig {
+    cassandra: {
+        hosts: string[],
+        username: string
+        password: string
+        keyspace: string,
+        datacenter: string
+    }
+    storageConfig: {
+        refreshInterval: number
+    }
+}
+
+export class StoragePlugin extends Plugin<StoragePluginConfig> {
+
+    private cassandra: Storage|undefined
+    private storageConfig: StorageConfig|undefined
+
+    constructor(options: PluginOptions) {
+        super(options)
+    }
+
+    async start() {
+        this.cassandra = await this.getCassandraStorage()
+        this.storageConfig = await this.createStorageConfig()
+        this.storageConfig.getStreams().forEach((stream) => {
+            this.subscriptionManager.subscribe(stream.id, stream.partition)
+        })
+        this.storageConfig.addChangeListener({
+            onStreamAdded: (stream: StreamPart) => this.subscriptionManager.subscribe(stream.id, stream.partition),
+            onStreamRemoved: (stream: StreamPart) => this.subscriptionManager.unsubscribe(stream.id, stream.partition)
+        })
+        this.networkNode.addMessageListener((msg) => {
```

Comment:
If the plugin is stopped, does this still fire?

Reply:
Also note IIUC this messageListener function is never removed, nor can it be with the current public interface, there's no `networkNode.removeMessageListener`, thus this instance can never be garbage collected until the node is GC'ed.

Reply:
Good catch 👍 Added cleanup logic to `Broker` and `StoragePlugin`: 

- stops the HTTP server before the plugin stops
- removes the message listener (added `networkNode.removeMessageListener `to `network` package)
- unsubscribes from each storable stream
- unsubscribes from the `storage-node-assignments` stream

Added also` StoragePlugin.test.ts` to test the cleanup logic.